openapi-bal-service/src/main/java/io/ballerina/openapi/converter/utils/ConverterCommonUtils.java

```diff
@@ -479,24 +481,20 @@ public static String getOpenApiFileName(String servicePath, String serviceName,
             openAPIFileName = serviceName.replaceAll(SLASH, "_");
         }
 
-        return getNormalizeFileName(openAPIFileName) + Constants.OPENAPI_SUFFIX +
+        return getNormalizedFileName(openAPIFileName) + Constants.OPENAPI_SUFFIX +
                 (isJson ? JSON_EXTENSION : YAML_EXTENSION);
     }
 
     /**
      * Remove special characters from the given file name.
      */
-    public static String getNormalizeFileName(String openAPIFileName) {
+    public static String getNormalizedFileName(String openAPIFileName) {
 
         String[] splitNames = openAPIFileName.split("[^a-zA-Z0-9]");
-        StringBuilder normalizeFileName = new StringBuilder();
-        for (String name: splitNames) {
-            if (!name.isBlank()) {
-                normalizeFileName.append(name).append(UNDERSCORE);
-            }
-        }
-        if (normalizeFileName.toString().endsWith(UNDERSCORE)) {
-            return normalizeFileName.deleteCharAt(normalizeFileName.length() - 1).toString();
+        if (splitNames.length > 0) {
+            return Arrays.stream(splitNames)
+                    .map(namePart -> namePart)
+                    .collect(Collectors.joining(UNDERSCORE));
```

Comment:
Can be simplified as `return String.join(UNDERSCORE, splitNames)`

Reply:
This `splitNames` array contains some time blank items, Therefor we need to add some item validation before joining the string :) .

Reply:
@lnash94 I don't see that blank string validation in here :) Then shall we stick to the steam APIs update the logic accordingly?

Reply:
Please refer to this commit https://github.com/ballerina-platform/openapi-tools/commit/e8c9a6dc5bbf752e40c68af2ea2ef5c6a1b38ed1 🙂